components/devicecloud-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/GroupManagerService.java

```diff
@@ -0,0 +1,252 @@
+    @Produces("application/json")
+    public boolean addNewSharingRoleForGroup(@FormParam("username") String username, @PathParam("groupId") int groupId, @FormParam("role") String roleName, @FormParam("permissions") String[] permissions) {
+        try {
+            Permission[] perms = new Permission[permissions.length];
+            for (int i = 0; i < permissions.length; i++) {
+                perms[i] = new Permission("group-mgt/user", permissions[i]);
+            }
+            return new GroupManagement().getGroupManagementService().addNewSharingRoleForGroup(username, groupId, roleName, perms);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}/role")
+    @DELETE
+    @Consumes("application/json")
+    @Produces("application/json")
+    public boolean removeSharingRoleForGroup(@PathParam("groupId") int groupId, @FormParam("role") String roleName) {
+        try {
+            return new GroupManagement().getGroupManagementService().removeSharingRoleForGroup(groupId, roleName);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    @Path("/group/id/{groupId}/role/all")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public String[] getAllRolesForGroup(@PathParam("groupId") int groupId) {
+        try {
+            List<String> roles = new GroupManagement().getGroupManagementService().getAllRolesForGroup(groupId);
+            String[] rolesArray = new String[roles.size()];
+            return roles.toArray(rolesArray);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Path("/group/id/{groupId}/{user}/role/all")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public String[] getGroupRolesForUser(@PathParam("user") String user, @FormParam("username") String username, @PathParam("groupId") int groupId) {
+        try {
+            List<String> roles = new GroupManagement().getGroupManagementService().getGroupRolesForUser(user, groupId);
+            String[] rolesArray = new String[roles.size()];
+            return roles.toArray(rolesArray);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Path("/group/id/{groupId}/user/all")
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    public String[] getUsersForGroup(@PathParam("groupId") int groupId) {
+        try {
+            List<String> users = new GroupManagement().getGroupManagementService().getUsersForGroup(groupId);
+            String[] usersArray = new String[users.size()];
+            return users.toArray(usersArray);
+        } catch (GroupManagementException e) {
+            e.printStackTrace();
```

Comment:
Ack, Already done in latest repository. Code moved to https://github.com/wso2/carbon-device-mgt